# Type: docs
# Subtype: D2
# English Summary: Commented in PasswordComparisonAuthenticator's Javadoc that it doesn't support SSHA passwords
# Chinese Summary: 在PasswordComparisonAuthenticator的Javadoc中添加了注释，以指示它无法处理SSHA密码
# Entry Number: 10

# === Git Diff Content ===

diff --git a/ldap/src/main/java/org/springframework/security/ldap/authentication/PasswordComparisonAuthenticator.java b/ldap/src/main/java/org/springframework/security/ldap/authentication/PasswordComparisonAuthenticator.java
index dada87b39..c6fa530e1 100644
--- a/ldap/src/main/java/org/springframework/security/ldap/authentication/PasswordComparisonAuthenticator.java
+++ b/ldap/src/main/java/org/springframework/security/ldap/authentication/PasswordComparisonAuthenticator.java
@@ -38,6 +38,8 @@ import org.springframework.util.Assert;
  * <p>
  * If passwords are stored in digest form in the repository, then a suitable {@link PasswordEncoder}
  * implementation must be supplied. By default, passwords are encoded using the {@link LdapShaPasswordEncoder}.
+ * Note that compare operations will not work if salted-SHA (SSHA) passwords are used, as it is not possible to
+ * know the salt value which is a random byte sequence generated by the directory.
  *
  * @author Luke Taylor
  * @version $Id$
